packages/rum/src/domain/startRecordBridge.ts

```diff
@@ -0,0 +1,17 @@
+import { getEventBridge } from '@datadog/browser-core'
+import type { ViewContexts } from '@datadog/browser-rum-core'
+import type { BrowserRecord } from '../types'
+
+export function startRecordBridge(viewContexts: ViewContexts) {
+  const bridge = getEventBridge<'record', BrowserRecord>()!
+
+  return {
+    addRecord: (record: BrowserRecord) => {
+      // Get the current active view, not at the time of the record, aligning with the segment logic.
+      // This approach could potentially associate the record to an incorrect view, in case the record date is in the past (e.g. frustration records).
+      // However the risk is minimal. We could address the issue when potential negative impact are identified.
+      const view = viewContexts.findView()!
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏In order to align with segment logic, what about:
- maintaining a "current context" for the record bridge, attached to the events
- creating it with the first added record
- clearing it on VIEW_CREATED or PAGE_EXITED

It would avoid to touch the curent logic around view contexts and view lifecycle.

Reply:
Maybe a "record context" module instead, replacing the segment context and used to get the view information for replay stats and bridge.

Reply:
Hum, the segment context is a concept shared with Replay that leaves in the [event format](https://github.com/DataDog/rum-events-format/blob/389581be98dcf8efbfcfe7bffaa32d53f960fb6f/lib/cjs/generated/browserSessionReplay.d.ts#L309). I think it might be confusing to replace it with a "record context"

Reply:
Resolved: https://github.com/DataDog/browser-sdk/pull/2598